### PR TITLE
[ROCm] Adding support to rnn ops

### PR DESCRIPTION
**Pull request by jerryyin**
Add ROCm support to core/kernels/rnn/*. The code change is trivial, please review and merge. @whchung @chsigg 

------
Test performed:

//tensorflow/contrib/rnn:lstm_ops_test
//tensorflow/contrib/rnn:gru_ops_test
//tensorflow/contrib/rnn:fused_rnn_cell_test
//tensorflow/contrib/rnn:rnn_test

### CODE REVIEW

**Review by chsigg — DISMISSED** [review]

**Comment by jerryyin** [comment]
@chsigg In the most recent commit, I manually updated the bazel build file's formatting so that buildifier does not complain. This is to address the `Ubuntu Sanity` build's complaint in `//tensorflow/tools/ci_build:gen_ci_sanity_out` target.

I wonder if there's a way to build this target locally? Since there doesn't seem to be a Build file that specifies `gen_ci_sanity_out`, I have to pull build and run buildifier all manually.

**Comment by jerryyin** [comment]
@chsigg A gentle ping

File: tensorflow/core/kernels/rnn/BUILD [hunk 1]
```diff
@@ -19,7 +23,11 @@ licenses(["notice"])  # Apache 2.0
 
 tf_gpu_library(
     name = "blas_gemm",
-    srcs = if_cuda_is_configured(["blas_gemm.cc"]),
+    srcs = [] + if_cuda_is_configured([
```
Comment (chsigg) [inline tensorflow/core/kernels/rnn/BUILD]: `[] +` looks unnecessary.

**Review by chsigg — APPROVED** [review]

File: tensorflow/core/kernels/rnn/BUILD (hunk 1 above)
Comment (jerryyin) [reply tensorflow/core/kernels/rnn/BUILD]: Agree with that. Though if I don't have it, `buildifier` will instead format it as the following:
```
    src = if_cuda_is_configured(["blas_gemm.cc"]) + if_rocm_is_configured([
          "blas_gemm.cc"
    ]),
```
Which is more ugly than how it looks now.

**Pull request merged**